feature: add example app, tweaks to webhooks docs (#558)
knocklabs/docs: content/guides/customer-webhooks.mdx

### Debugging failed webhook deliveries

> You can use the [Message delivery logs API](https://docs.knock.app/reference#get-message-delivery-logs) to create UI in your application that lets developers inspect the request/response interaction between Knock and their downstream URL.

Reviewer: I wonder if this is more a both/and sort of thing? These are available via API but you can also debug with dashboard logs (which I'm assuming is what the original is referencing). We could potentially point to https://docs.knock.app/send-notifications/debugging-workflows?

Author: Yeah, that's fair, I updated this language to include the original text and a link to that URL, but also kept this addition with a framing like this: 'Since your customers won't have access to Knock, you can use the APIs to create UI for them...'

Reviewer: perf

### Reporting on failed webhook deliveries

You might want to provide a way to notify your customers that their webhooks are failing. You can do so by leveraging Knock’s [outbound webhooks](https://docs.knock.app/send-and-manage-data/outbound-webhooks) features to create a webhook callback to your server that listens for `message.undelivered` events.

### Providing a dynamic signing key

> You will likely want to provide a way for your customers to [verify that the webhook request is coming from Knock](/integrations/webhook/overview#securing-your-webhooks). You can do so by providing a signing key in the webhook configuration and Knock will use it to sign the request with a HMAC/SHA256 signature. For customer-facing webhooks, you can provide a UI for your customers to configure the signing key and you'll want to configure this as [a dynamic signing key](/integrations/webhook/overview#using-a-dynamic-signing-key) using Liquid variables.

Reviewer: Couple small nits

```suggestion
You will likely want to provide a way for your customers to [verify that the webhook request is coming from Knock](/integrations/webhook/overview#securing-your-webhooks). You can do so by providing a signing key in your webhook channel's configuration that Knock will use to sign the request with an HMAC/SHA256 signature.

For customer-facing webhooks, you can provide a UI for your customers to configure a signing key; you'll want to configure this in Knock as a [dynamic signing key](/integrations/webhook/overview#using-a-dynamic-signing-key) using Liquid variables.
```
